### PR TITLE
Change copyright

### PR DESCRIPTION
**Pull request by mPokornyETM**
# Request for a change on public repository on [CtrlppCheck](https://github.com/siemens/CtrlppCheck)

## Basic information
This changes provide automatic way to change Copyright entries for whole ctrl-code (inclusive tests)

**please note** that as first we need merge this changes #44

### Testing done

The result can you see in changed files.
Tests after change are also OK:
image

### Proposed upgrade guidelines

N/A

### Localizations

N/N

### Submitter checklist

~~- [ ] The Github issue, if it exists, is well-described.~~
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood.
  - The changelog generator for plugins uses the **pull request title as the changelog entry**.
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during the upgrade.
~~- [ ] There is automated testing or an explanation that explains why this change has no tests.~~
~~- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.~~
~~- [ ] Any localizations are transferred to /msg/ files.~~
- [x] Automated tests has been executed and valid
- [x] Changes in the interface are documented

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] Changelog entries in the **pull request title** and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] C++ and Control(++) code changes are tested by automated test.
- [ ] WinCC OA guidelines for C++ and Control(++) coding have been met.
- [ ] Result of pipeline build proving error/warning free code.
- [ ] Result of automatic tests proving regression free.

### CODE REVIEW

File: WinCC_OA_Test/executeTests.cmd [hunk 1]
```diff
@@ -80,6 +92,20 @@ IF %ERRORLEVEL% NEQ 0 IF %ERRORLEVEL% NEQ 3 (
...
 
+REM --------------------------------------------------------------------------
+REM Format ctrl code
+if %FORMAT_CTRL_CODE% == true (
+  call %oaBinPath%WCCOActrl.exe -config %WINCC_OA_TEST_PATH%Projects\TfCustomizedQG\config\config -n astyle.ctl %WINCC_OA_TEST_PATH% -log +stderr -lang en_US.utf8
+  call %oaBinPath%WCCOActrl.exe -config %WINCC_OA_TEST_PATH%Projects\TfCustomizedQG\config\config -n astyle.ctl %WINCC_OA_TEST_PATH%..\WinCCOA_QualityChecks -log +stderr -lang en_US.utf8
+)
+
+REM --------------------------------------------------------------------------
+REM Chnage Copyright information
```
Comment (dhoegerlETM) [inline WinCC_OA_Test/executeTests.cmd:103]:
```suggestion
REM Change copyright information
```

File: WinCC_OA_Test/readme.md [hunk 2]
```diff
@@ -18,7 +18,9 @@ Following options are possible:
 -OaVersion ,defines the WinCC OA Version (default 3.20)
 -OaInstallPath ,define the installation path of WinCC OA (default C:\Siemens\Automation\WinCC_OA\)
 -OaTestPath ,define the test path (default *thisWorkspace*\WinCC_OA_Test\)
--OaTestRunId , defines test-run ID to be executed (default Regression-tests )
+-OaTestRunId , defines test-run ID to be executed (default Regression-tests)
+-formatCtrlCode, allow automatic formatting of ctrl code - inclusive ctrl tests (default disabled)
+-changeCopyright, change copyright entries to correct year for whole ctrl code - inclusive ctrl tests (default disabled)
```
Comment (dhoegerlETM) [inline WinCC_OA_Test/readme.md:23]:
```suggestion
-changeCopyright, change copyright entries to correct year for all ctrl code - including ctrl tests (default disabled)
```

File: WinCC_OA_Test/Projects/TfCustomizedQG/scripts/copyright.ctl [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      result = "//--------------------------------------------------------------------------------\n" +
+               "/**\n  @file $relPath\n  @copyright " + correctCopyright +
+               "\n             SPDX-License-Identifier: GPL-3.0-only\n*/\n\n" +
+               result;
+      throwError(makeError("", PRIO_INFO, ERR_CONTROL, 0, "New copyright added", ctrlFile.getPath()));
+      changed = true;
+    }
+
+    if (changed && (ctrlFile.write(result) != 0))
+      throwError(makeError("", PRIO_INFO, ERR_CONTROL, 0, "Can not change file", ctrlFile.getPath()));
```
Comment (dhoegerlETM) [inline WinCC_OA_Test/Projects/TfCustomizedQG/scripts/copyright.ctl:68]:
```suggestion
      throwError(makeError("", PRIO_INFO, ERR_CONTROL, 0, "Cannot change file", ctrlFile.getPath()));
```

**Review by dhoegerlETM — CHANGES_REQUESTED** [review]
Generally speaking: why include the functions "updateCopyright" and "formatCode" directly into the executeTests.cmd script? I would appreciate to have an own e.g. "handleCtrlCode" script for those functions.

**Comment by mPokornyETM** [comment]
> Generally speaking: why include the functions "updateCopyright" and "formatCode" directly into the executeTests.cmd script? I would appreciate to have an own e.g. "handleCtrlCode" script for those functions.

I am inspired by mvn usage (Java) and want to spare duplicated lines (or complex batcscripts struct). Whwn we want provide more solutions here, I will care about that.

**Comment by dhoegerlETM** [comment]
> > Generally speaking: why include the functions "updateCopyright" and "formatCode" directly into the executeTests.cmd script? I would appreciate to have an own e.g. "handleCtrlCode" script for those functions.
> 
> I am inspired by mvn usage (Java) and want to spare duplicated lines (or complex batcscripts struct). Whwn we want provide more solutions here, I will care about that.

Understand, but then we must rename the tool from executeTests.cmd to something more general, e.g. executeCommands or batchAutomate or executeBatch.

**Comment by mPokornyETM** [comment]
> > > Generally speaking: why include the functions "updateCopyright" and "formatCode" directly into the executeTests.cmd script? I would appreciate to have an own e.g. "handleCtrlCode" script for those functions.
> > 
> > 
> > I am inspired by mvn usage (Java) and want to spare duplicated lines (or complex batcscripts struct). Whwn we want provide more solutions here, I will care about that.
> 
> Understand, but then we must rename the tool from executeTests.cmd to something more general, e.g. executeCommands or batchAutomate or executeBatch.

OK, I will thing about that, and provide some better name.

**Comment by mPokornyETM** [comment]
As discussed there is new PR where the tools has been splitted. Therefor may be this one closed

**Comment by mPokornyETM** [comment]
Only to be precise: See #47

**Pull request closed without merging**